recipe/patch_yaml/libxml2.yaml

```diff
@@ -46,3 +48,16 @@ then:
   - tighten_depends:
       name: libxml2
       upper_bound: 2.14.0
+---
+if:
+  timestamp_lt: 1744044827000
+  # for anything not caught by the above, ensure that packages (built before 2.14)
+  # which assumed major-level compatibility get the right upper bound too
+  has_depends:
+    - libxml2 *<3.0a0
+  not_has_depends:
+    - libxml2 >=2.14*
```

Comment:
We'd want to do the same for packages already built against `>=2.14*` (but with upper_bound 2.15). I've left this out here to reduce the number of affected artefacts